### DISCUSSION TITLE
Question about contract indexes

### DISCUSSION THREAD

**Discussion by oacdutra** [RGB-WG/rgb#54] [Q&A] [answered]
Hi @dr-orlovsky,

I tried pay a rgb invoice using `rgb cli`, but I receive the following error:

```
[2023-03-30T04:51:09Z DEBUG rgb] Executing command: transfer
thread 'main' panicked at 'error paying invoice: Consigner(InventoryError(InternalInconsistency(Stash(ContractAbsent(ContractId(Array<32>(0e366c92ad6f0dc1c0ae223aacd8b258aef8f2b72bcbd95208449ab205eb3df2)))))))', src/bin/rgb/command.rs:411:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

After investigating, I found the reason: 

Into the `consign` method, we tried to get contract_index information (see [here](https://github.com/RGB-WG/rgb-wallet/blob/36d4a451cc61eb5ffc978c272daa80f6177b9435/std/src/persistence/inventory.rs#L419)). However, in no previous step did we fill in this information. 

Unfortunately, I'm unsure if I understand the contract indexes purpose. Could you explain the objective of the contract indexes for me to try to complete this step, please?

Thanks a lot!

**Comment by dr-orlovsky** [comment, marked as answer]
This is a clear bug. I am still debugging the beta released a day before yesterday, so I expect there might be other bugs

**Reply by oacdutra** [reply]
> This is a clear bug. I am still debugging the beta released a day before yesterday, so I expect there might be other bugs

Thanks for the answer.

**Reply by dr-orlovsky** [reply]
Can you pls convert this jnto an issue?
